### ISSUE TITLE
Focus editor and position cursor at the right place after custom undo/redo

### ISSUE THREAD

**Issue by Vadorequest** [quilljs/quill#625]
Hello,
We have implemented our own undo/redo but the cursor isn't correctly placed once the undo/redo action is finished. How can I move the cursor at the right position once the action is done? Why does the cursor moves at the beginning of the line? (just so I understand)

**Expected behavior**: 
When we click on the undo/redo button, the action is applied and the cursor moves at the right position depending on the action.

**Actual behavior**: 
The cursor is moved at the beginning of the line after the action is done.

**Version**: 0.20

Thanks :)

**Comment by yuchentseng** [comment]
I found that the issue is the cursor will moves at the beginning of the line whenever I press "any toolbar action".

**Comment by Vadorequest** [comment]
@yuchentseng Yeah, it is what I'm experiencing as well, excepted that I'm also using a custom toolbar.
Anyway, it would be nice to have a solution for this, for a better user experience.

**Comment by jhchen** [comment]
Are you restoring the selection in your custom undo/redo manager?

**Comment by Vadorequest** [comment]
What do you mean ?
The undo/redo just changes the text value. It doesn't do any text selection.

**Comment by jhchen** [comment]
On most text editors if you type something, then move your cursor somewhere else, then undo, the cursor moves back to where the text changed. So yes undo managers must handle cursor/selections as well. Quill's undo manager does this so if you are extending it you must either not interfere with this part of it or handle it yourself as well.

**Comment by Vadorequest** [comment]
Could you show me where Quill does this? Because it seems we tamper with it (I don't see why we would loose the cursor otherwise), so I'd like to implement it and that would be great to have an example. ;)

**Comment by jhchen** [comment]
https://github.com/quilljs/quill/blob/master/src/modules/undo-manager.coffee#L100

**Comment by Vadorequest** [comment]
Thanks, we solved it by re-selecting the text on `shouldComponentUpdate` (React) . :)
